### PR TITLE
Add Division Operation

### PR DESCRIPTION
**Pull request by fdeligoez**
## What's Changed
- Implemented divide() function with:
  * Zero division protection
  * Type hints for better code clarity
  * Unit tests including edge cases

## Testing
✓ Basic division operations
✓ Zero division check
✓ Negative numbers
✓ Decimal results

## Notes
- Added ValueError for zero division cases
- Full test coverage including error cases

### CODE REVIEW

File: src/calculator.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+def multiply(a: float, b: float) -> float:
+    return a * b
+
+def divide(a: float, b: float) -> float:
```
Comment (fdeligoez) [inline src/calculator.py:4]: Good use of type hints! This makes the function signature very clear.

File: src/calculator.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+    if b == 0:
```
Comment (fdeligoez) [inline src/calculator.py:5]: Consider adding more descriptive error message like: 'Division by zero is not allowed' for better user understanding.
